src/app/actions/access.js

```diff
@@ -80,6 +80,12 @@ export const openWeb3Wallet = (selectedProvider, forwardUrl) => (dispatch) => {
       } else {
         message = `Error connecting to ${providerName}: ${message}`
       }
+      // wallet info modal if no web3 instance, if trust use deep linking
+      if (providerName !== 'Trust Wallet' && message !== 'Unsupported network') {
```

Comment:
This will try to redirect to trustwallet in cases you shouldn't be. Also, don't use the name in case we decide to change "Trust Wallet" to something else. `selectedProvider` is the identifier here.
```javascript
if (message !== 'Unsupported network') {
  if (selectedProvider !== 'trust') {
    dispatch(push(routes.connectWalletInfo(selectedProvider)))
  } else {
    window.location.href = 'https://links.trustwalletapp.com/SBr41u7nVR?&event=openURL&url=https://faa.st/app/connect'
  }
}
```